### ISSUE TITLE
Cannot fine-tune google/mobilebert-uncased using native Pytorch

### ISSUE THREAD

**Issue by jeffreyyu0602** [huggingface/transformers#15050]
## Environment info
- `transformers` version: 4.12.5
- Platform: Linux-4.4.0-210-generic-x86_64-with-glibc2.23
- Python version: 3.9.7
- PyTorch version (GPU?): 1.10.0 (False)
- Tensorflow version (GPU?): 2.4.1 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help
@LysandreJik  @sgugger

## To reproduce

1. Follow the "fine-tuning in native PyTorch" section in "Fine-tuning a pretrained model" on the official documentation
2. Replace "bert-base-cased" with "google/mobilebert-uncased"
3. Add max_length=171 argument to tokenize function

## Expected behavior

The model fine-tuned with native PyTorch reach the same accuracy as the model fine-tuned with a Trainer. But instead the accuracy stucks around 50% and even dropped after one epoch.

**Comment by NielsRogge** [comment]
Hi,

For training-related questions, please refer to the forum. See [here](https://discuss.huggingface.co/search?q=mobilebert) for all MobileBERT-related questions for instance.

We like to keep Github issues for bugs/feature requests.

Thank you!
